### PR TITLE
Add basic setup

### PR DESCRIPTION
**Pull request by kajalsinha**
add basic golang setup with react, auth0.

part of structure took from https://github.com/madhanganesh/taskpad

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,5 @@
+POSTGRES_CONNECTION_STRING=postgresql://postgres:postgres@localhost/jobsgowhere?sslmode=disabled
+AUTH0_DOMAIN=jobsgowhere.au.auth0.com
+AUTH0_CLIENT_ID=rc3Waw3xebsUoG2uE7I2Zb1T5jgSfrTF
```
Comment (subhransu) [inline .env:3]: let's move this to environment variables?

File: api/middlewares/auth-middleware.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+	jwtmiddleware "github.com/auth0/go-jwt-middleware"
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/gin-gonic/gin"
+)
+
+// JSONWebKeys struct
+type JSONWebKeys struct {
+	Kty string   `json:"kty"`
+	Kid string   `json:"kid"`
```
Comment (subhransu) [inline api/middlewares/auth-middleware.go:19]: these short keywords are common in go? I would normally prefer full words.

**Review by subhransu — CHANGES_REQUESTED** [review]
Don't have many comments at the moment other than a few clarification. But let's not use `Auth0`. 
We can just use LinkedIn Authentication: https://developer.linkedin.com/docs/guide/v2/concepts/authentication#auth

@lxcid @ivanfoong do you have any comments?

File: .env (hunk 1 above)
Comment (kajalsinha) [reply .env:3]: in go, these are env vars. later we can put it in ci/cd.

File: api/middlewares/auth-middleware.go (hunk 2 above)
Comment (kajalsinha) [reply api/middlewares/auth-middleware.go:19]: Yes, these words are too short, I too prefer a little description there in naming. But these are coming from JSON web token RFC. It helps to map with RFC model.

More details here https://tools.ietf.org/html/rfc7517. 

The RFC is kind of one time thing, and won't be that frequent. 
We will use full words for the api implementations.

File: .env (hunk 1 above)
Comment (subhransu) [reply .env:3]: we should add this to `.gitignore` right?

Comment (ivanfoong) [reply .env:3]: Are we targeting more white collar population? Or general population? I feel that there's a better chance of having gmail and using google account

File: api/controllers/task-controller.go [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+
+}
+
+// CreateTask method
+func (c *TaskController) CreateTask(ctx *gin.Context) {
+	var task models.Task
+	ctx.BindJSON(&task)
+	if task.Title == "" {
+		ctx.JSON(400, gin.H{
+			"error": "title should not be empty",
```
Comment (ivanfoong) [inline api/controllers/task-controller.go:35]: Should we include `error_code` too? for easy error handling/matching in FE

**Review by ivanfoong — COMMENTED** [review]
Should we have `error_code` field in json for every error response? so it's easy for FE to do error handling/matching

**Comment by lxcid** [comment]
@kajalsinha I hope you don't mind I recreate the UI side with CRA's TypeScript. I believe the ui is build from a project that uses CRA. It come with quite a few boilerplate that I foresee us not needing it.

Let me know if you have thoughts on this. Will send you a PR that merge to your branch.

Edited:

Check out https://github.com/jobsgowhere/jobsgowhere/pull/10

**Comment by subhransu** [comment]
@lxcid @ivanfoong is this first PR good to be merged for now? 
@lxcid I have seen another PR from you ... that goes after this?

**Review by subhransu — APPROVED** [review]

**Comment by ivanfoong** [comment]
Good to merge, I have merged in Stan pr

**Comment by lxcid** [comment]
Can I request a docker-compose setup for development? I don't have Go installed in my machine.

**Pull request merged**